vms/platformvm/state/staker.go

```diff
@@ -83,27 +83,38 @@ func (s *Staker) Less(than *Staker) bool {
 	return bytes.Compare(s.TxID[:], than.TxID[:]) == -1
 }
 
-func NewCurrentStaker(txID ids.ID, staker txs.Staker, potentialReward uint64) (*Staker, error) {
+func (s *Staker) Duration() time.Duration {
+	return s.EndTime.Sub(s.StartTime)
+}
+
+func NewCurrentStaker(
+	txID ids.ID,
+	staker txs.Staker,
+	startTime time.Time,
+	potentialReward uint64,
+) (*Staker, error) {
 	publicKey, _, err := staker.PublicKey()
 	if err != nil {
 		return nil, err
 	}
-	endTime := staker.EndTime()
+
+	stakingDuration := staker.Duration()
+	endTime := startTime.Add(stakingDuration)
```

Comment:
This will change with continuous staking but it's correct for pending stakers removal